noir-projects/noir-protocol-circuits/crates/types/src/merkle_tree/variable_merkle_tree.nr

```diff
@@ -33,7 +33,11 @@ impl VariableMerkleTree {
     //
     pub fn new_sha<let N: u32>(leaves: [Field; N], num_non_empty_leaves: u32) -> Self {
         // Find size of tree required
-        let height = unsafe { get_height(num_non_empty_leaves, 0) };
+        let height = unsafe {
+            //@safety This is safe because we derive next and previous powers of 2 below based on the value
+            // and the powers are then checked against the num non empty leaves.
```

Comment:
The comment is right, but it shows how this function is awful. There should be a fn that takes `num_non_empty_leaves` and returns `prev_power_2`, where this is done inside of that. Otherwise it's hard to check how the unconstrained value is used.

Reply:
Good idea. Will address in a PR up the stack.